tools/make-kani-release/src/main.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This file is a glorified shell script for constructing a Kani release bundle.
+//! We use Rust here just to aid in making the "script" more robust.
+//!
+//! Run with `cargo run -p make-kani-release -- <version>` and this will produce
+//! (e.g.) `kani-1.0-x86_64-unknown-linux-gnu.tar.gz`.
+
+use std::{ffi::OsString, path::Path, process::Command};
+
+use anyhow::{bail, Result};
+
+fn main() -> Result<()> {
+    let version_string = parse_args()?;
+    let kani_string = format!("kani-{}", version_string);
+    let bundle_name = format!("{}-{}.tar.gz", kani_string, env!("TARGET"));
+
+    if !Path::new("src/kani-compiler").exists() {
+        bail!("Run from project root directory. Couldn't find 'src/kani-compiler'.");
+    }
+
+    let dir = Path::new(&kani_string);
+    if dir.exists() {
+        bail!(
+            "Directory {} already exists. Previous failed run? Delete it first.",
+            dir.to_string_lossy()
+        );
+    }
+    std::fs::create_dir(dir)?;
+
+    bundle_kani(dir)?;
+    bundle_cbmc(dir)?;
+    // We don't bundle viewer, and we instead `pip install` it on-first run,
+    // due to issues with portability between python versions the user may have.
+    // bundle_viewer(dir)?;
+
+    create_release_bundle(dir, &bundle_name)?;
+
+    std::fs::remove_dir_all(dir)?;
+
+    println!("\nSuccessfully built release bundle: {}", bundle_name);
+    Ok(())
+}
+
+/// Parse command line arguments, and return the only thing we expect: a version string
+fn parse_args() -> Result<String> {
+    let args: Vec<_> = std::env::args().collect();
+    if args.len() != 2 {
+        bail!("Usage: cargo run -p make-kani-release -- <version>");
+    }
+    Ok(args[1].clone())
+}
+
+/// Copy Kani files into `dir`
+fn bundle_kani(dir: &Path) -> Result<()> {
+    let bin = dir.join("bin");
+    std::fs::create_dir(&bin)?;
+
+    // 1. Kani binaries
+    Command::new("cargo").args(&["build", "--release"]).run()?;
+
+    let release = Path::new("./target/release");
+    cp(&release.join("cargo-kani"), &bin)?;
```

Comment:
What about standalone kani?

Reply:
That's part of the shim/proxy, not the release bundle. Both invoke this `cargo-kani` binary just in different modes.

Reply:
I thought you had mentioned the idea of creating two different binaries. Did you give up on that idea?

Reply:
Right, that's the shim/proxy that calls this binary.